tests/SimplePacker.hpp

```diff
@@ -0,0 +1,292 @@
+    //! \param data The variable to unpack
+    //! \param buffer Buffer to unpack from
+    //! \param position Position in buffer to use
+    static void unpack(T& data,
+                       std::vector<char>& buffer,
+                       int& position)
+    {
+        unpack(&data, 1, buffer, position);
+    }
+
+    //! \brief Unpack an array of POD.
+    //! \param data The array to unpack
+    //! \param n Length of array
+    //! \param buffer Buffer to unpack from
+    //! \param position Position in buffer to use
+    static void unpack(T* data,
+                       std::size_t n,
+                       std::vector<char>& buffer,
+                       int& position)
+    {
+        std::memcpy(data, &buffer[position], n*sizeof(T));
+        position += n*sizeof(T);
+    }
+};
+
+//! \brief Default handling for unsupported types.
+template<class T>
+struct Packing<false,T>
+{
+    static std::size_t packSize(const T&)
+    {
+        static_assert(!std::is_same_v<T,T>, "Packing not supported for type");
+        return 0;
+    }
+
+    static void pack(const T&, std::vector<char>&, int&)
+    {
+        static_assert(!std::is_same_v<T,T>, "Packing not supported for type");
+    }
+
+    static void unpack(T&, std::vector<char>&, int&)
+    {
+        static_assert(!std::is_same_v<T,T>, "Packing not supported for type");
+    }
+};
+
+//! \brief Specialization for std::bitset
+template <std::size_t Size>
+struct Packing<false,std::bitset<Size>>
+{
+    static std::size_t packSize(const std::bitset<Size>& data)
+    {
+        return Packing<true,unsigned long long>::packSize(data.to_ullong());
+    }
+
+    static void pack(const std::bitset<Size>& data,
+                     std::vector<char>& buffer, int& position)
+    {
+        Packing<true,unsigned long long>::pack(data.to_ullong(), buffer, position);
+    }
+
+    static void unpack(std::bitset<Size>& data,
+                       std::vector<char>& buffer, int& position)
+    {
+        unsigned long long d;
+        Packing<true,unsigned long long>::unpack(d, buffer, position);
+        data = std::bitset<Size>(d);
+    }
+};
+
+template<>
+struct Packing<false,std::string>
+{
+    static std::size_t packSize(const std::string& data)
+    {
+        return sizeof(std::size_t) + data.size();
+    }
+
+    static void pack(const std::string& data,
+                     std::vector<char>& buffer, int& position)
+    {
+        Packing<true,std::size_t>::pack(data.size(), buffer, position);
+        Packing<true,char>::pack(data.data(), data.size(), buffer, position);
+    }
+
+    static void unpack(std::string& data, std::vector<char>& buffer, int& position)
+    {
+        std::size_t length = 0;
+        Packing<true,std::size_t>::unpack(length, buffer, position);
+        std::vector<char> cStr(length+1, '\0');
+        Packing<true,char>::unpack(cStr.data(), length, buffer, position);
+        data.clear();
+        data.append(cStr.data());
```

Comment:
You probably want to use
```C++
data.append(cStr.data(), cStr.size());
```
here just to avoid potential issues if `cStr` has any embedded nuls.  Of course those can't happen in well-formed UTF-8, but it's better to be safe than sorry.

Reply:
done.